### PR TITLE
Property replication test

### PR DESCRIPTION
**Pull request by oblm**
#### Description
This is a functional test. It verifies that an Actor can replicate a property across the network during play. It contains 1 Server and 2 Client workers.

The flow of the test is as follows:
 **Setup:**
 * The Server spawns one `ReplicatedTestActor`.

 **Test:**

 *  Both Clients check that they can see exactly 1 `ReplicatedTestActor`.
 *  The Server changes the `ReplicatedProperty` of the `ReplicatedTestActor` from `0` to `99`.
 *  The Clients check that the `ReplicatedProperty` is now set to `99`.

**Clean-up:**

 *  `ReplicatedTestActor` is destroyed using the `RegisterAutoDestroyActor` helper function.

#### Release note
Not applicable.

#### Tests
* This is a test.
* I ran it 100 times in a row to check that it works consistently and repeatedly.
* I also ran it as part of running the Functional Test Suite, this ensure that the tests that run before and after it clean up after themselves and do not impact on this test.

#### Documentation
I detailed the process of writing this test in: https://brevi.link/how-to-test-unrealgdk

#### Dependencies
This PR depends on this PR: https://github.com/spatialos/UnrealGDKTestGyms/pull/318

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestPropertyReplication/SpatialTestPropertyReplication.cpp
Comment (mironec) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestPropertyReplication/SpatialTestPropertyReplication.cpp:14]:
```suggestion
 * - Test:
```
Ultra nit

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestPropertyReplication/SpatialTestPropertyReplication.cpp
Comment (mironec) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestPropertyReplication/SpatialTestPropertyReplication.cpp:25]: Wanted to get your opinion on this as well.
Author field yay or nay?

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestPropertyReplication/SpatialTestPropertyReplication.cpp
Comment (mironec) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestPropertyReplication/SpatialTestPropertyReplication.cpp:47]: Hmmm, we have some new functions you could use in this kind of situation...
Can go over it tomorrow

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestPropertyReplication/SpatialTestPropertyReplication.cpp
Comment (oblm) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestPropertyReplication/SpatialTestPropertyReplication.cpp:25]: Is this question aimed at me? If so I'm pro because I can easily identity who wrote the test and owns maintaining it when it falls over.

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestPropertyReplication/SpatialTestPropertyReplication.cpp
Comment (oblm) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestPropertyReplication/SpatialTestPropertyReplication.cpp:47]: Sure thing, let's pair.

Comment (oblm) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestPropertyReplication/SpatialTestPropertyReplication.cpp:50]: added "Requires"

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestPropertyReplication/SpatialTestPropertyReplication.cpp
Comment (mironec) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestPropertyReplication/SpatialTestPropertyReplication.cpp:35]: Any reason for 100, 100, 80? Usually I would just do something at 0, 0.

Comment (oblm) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestPropertyReplication/SpatialTestPropertyReplication.cpp:35]: > Any reason for 100, 100, 80? Usually I would just do something at 0, 0.

I think I inherited this from another test I pinched from. I can set it to `0.0f, 0.0f, 0.0f` is prefereable?

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestPropertyReplication/SpatialTestPropertyReplication.cpp
Comment (mironec) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestPropertyReplication/SpatialTestPropertyReplication.cpp:8]: Maybe worth a comment explaining that this is an example test?

Comment (mironec) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestPropertyReplication/SpatialTestPropertyReplication.cpp:8]: / exemplary test

**Review by mironec — APPROVED** [review]
A few nits, but approving.
The test should go in first and then the TestGyms map.

**Comment by oblm** [comment]
This test was written using UnrealEngine version `4.25`, which is currently the supported version of the Engine, with `4.24` being the legacy version.

This will soon™ change, with `4.26` becoming the newly supported version, `4.25` becoming the legacy version and `4.24` being depreated.

When this occurs, I can merge this PR, but if I do so before hand this test will break CI.

This PR is now blocked on: [UNR-4391](https://improbableio.atlassian.net/browse/UNR-4391) - Drop support for Unreal 4.24

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestPropertyReplication/SpatialTestPropertyReplication.cpp
Comment (thomas-yung) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestPropertyReplication/SpatialTestPropertyReplication.cpp:25]: I also am in favour of this but would suggest changing this/including to the owner's email instead
e.g. `Author = "Thomas Yung (thomasyung@improbable.io)"`

File: SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp
Comment (oblm) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp:419]:
```suggestion
```

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestPropertyReplication/SpatialTestPropertyReplication.cpp
Comment (oblm) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestPropertyReplication/SpatialTestPropertyReplication.cpp:26]:
```suggestion
	Author = "Ollie Balaam (oliverbalaam@improbable.io)";
```

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestPropertyReplication/SpatialTestPropertyReplication.cpp
Comment (oblm) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestPropertyReplication/SpatialTestPropertyReplication.cpp:27]:
```suggestion
	Description = TEXT("This tests that an Actor can replicate a property across the network during play. It is an example test intended to teach the basics of the UnrealGDK Functional Test Framework. It's accompanied by this document: https://brevi.link/how-to-test-unrealgdk");
```

**Comment by oblm** [comment]
Test ran as premerge and [passed](https://buildkite.com/improbable/unrealgdk-premerge/builds/23036#a481ec74-d8b4-4ab9-a2bf-79e4bc02860b/1463-3731).

**Pull request merged**
